src/components/AccountDrawer/MiniPortfolio/index.tsx

```diff
@@ -88,25 +93,52 @@ const Pages: Array<Page> = [
 
 export default function MiniPortfolio({ account }: { account: string }) {
   const isNftPage = useIsNftPage()
+  const theme = useTheme()
   const [currentPage, setCurrentPage] = useState(isNftPage ? 1 : 0)
+  const [activityUnread, setActivityUnread] = useState(false)
   const shouldDisableNFTRoutes = useAtomValue(shouldDisableNFTRoutesAtom)
 
-  const Page = Pages[currentPage].component
+  const { component: Page, key: currentKey } = Pages[currentPage]
+
+  const hasPendingTransactions = useHasPendingTransactions()
+
+  useEffect(() => {
+    if (hasPendingTransactions && currentKey !== 'activity') setActivityUnread(true)
+  }, [currentKey, hasPendingTransactions])
```

Comment:
i'll probably become "that guy" and its firmly in nit territory but whenever you are doing a simple effect like this you can save a few keystrokes and a bit of render performance:

```tsx
 if (hasPendingTransactions && currentKey !== 'activity' && !activityUnread) {
  setActivityUnread(true)
}
```

dan abramov told me so, but its not the biggest deal

https://react.dev/learn/you-might-not-need-an-effect

Reply:
lollll i love this, it does save the component from rerendering if the variable hasnt changed